soc/nordic/nrf54l/soc.c

```diff
@@ -109,13 +113,15 @@ static int nordicsemi_nrf54l_init(void)
 		nrf_power_task_trigger(NRF_POWER, NRF_POWER_TASK_CONSTLAT);
 	}
 
+#if defined(NRF_APPLICATION)
```

Comment:
I suppose whole `nordicsemi_nrf54l_init` can be APP-specific, except for `SystemCoreClockUpdate`.